### ISSUE TITLE
Implement pyp2rpm hook to get extra information when rebasing packages from PyPI

### ISSUE THREAD

**Issue by thozza** [rebase-helper/rebase-helper#226]
```pyp2rpm``` is able to provide metadata about the latest version of a package from PyPI. Such metadata include:
* list of runtime dependencies
* list of build dependencies
* list of files, modules, etc.

Rebase-helper could use such information to modify the rebased SPEC in order to increase the chance for successful RPM build. We could modify the [Build]Requires and %files section accordingly.

This should be implemented as a hook plugin AFTER creation of rebased SPEC file. This way we can solve the same thing also for other language ecosystems.

**Comment by mcyprian** [comment]
Metadata of the specific version of the package can be extracted using following classes:
```
class pyp2rpm.metadata_extractors.SetupPyMetadataExtractor(
    self, local_file, name, name_convertor, version, rpm_name=None, 
    venv=True, base_python_version=settings.DEFAULT_PYTHON_VERSION, 
    metadata_extension=False)
```
Parameters:
- local_file: path to source archive of the package
- name: name of the package
- name_convertor: instance of pyp2rpm's name_convertor 
                              (NameConvertor, DandifiedNameConvertor)
- version: version of the package

`class pyp2rpm.name_convertor.NameConvertor(self, distro)`
Parameters:
- distro: distro whose conversion rules to use

Example Usage:
```
In [1]: from pyp2rpm.name_convertor import NameConvertor
In [2]: from pyp2rpm.metadata_extractors import SetupPyMetadataExtractor
In [3]: nc = NameConvertor('fedora')
In [4]: extractor = SetupPyMetadataExtractor('/home/mcyprian/Downloads/simpleeval-0.8.7.tar.gz', 'simpleeval', nc, '0.8.7')

option1:
In [5]: extractor.runtime_deps
Out[5]: [['Requires', 'python-py', '>=', '1.4.29'], ['Requires', 'python-setuptools']]
In [6]: extractor.build_deps
Out[6]: [['BuildRequires', 'python-setuptools']]
In [7]: extractor.scripts
Out[7]: {'py.test', 'py.test-3.5'}
In [8]: extractor.py_modules
Out[8]: {'pytest'}
In [9]: extractor.packages
Out[9]: {'_pytest'}

option2:
In [5]: data = extractor.extract_data()
In [6]: data.data
Out[7]: 
{ 'runtime_deps': [['Requires', 'python-py', '>=', '1.4.29'],
                             ['Requires', 'python-setuptools']],
  'build_deps': [['BuildRequires', 'python2-devel'],
                        ['BuildRequires', 'python-setuptools'],
                        ['BuildRequires', 'python-sphinx']],
 'packages': {'_pytest'},
 'pkg_name': 'python-pytest',
 'py_modules': {'pytest'},
...
}

```

**Comment by skisela** [comment]
The idea looks great, but in case of rebasing to python-tox from 2.7.0 version to 2.9.1, one of the extracted BuildRequires is `['BuildRequires', 'python-pluggy', u'>=', u'0.3.0']`.

According to koji(https://koji.fedoraproject.org/koji/buildinfo?buildID=1038518, https://koji.fedoraproject.org/koji/buildinfo?buildID=940380)
f27 has packages: python-pluggy, python3-pluggy
f28: python2-pluggy, python3-pluggy

Am I right, that there is no checking, if the RPM package actually exists?

The problem I see here is, that there still are packages without the explicit `python2` version. I might be missing something important there, so please correct me in that case.

I guess it would make more sense to have the checking on `pyp2rpm` side, instead of checking it in `rebase-helper`.

Any ideas on checking the existence of the RPM package besides using `repoquery` or such others?

**Comment by mcyprian** [comment]
@skisela you are right, NameConvertor instance simply guess what the right name of the dependency might be. We also have [DandifiedNameConvertor](https://github.com/fedora-python/pyp2rpm/blob/master/pyp2rpm/name_convertor.py#L177), which does the search. The best solution for f25+ are [Automatic provides](https://fedoraproject.org/wiki/Changes/Automatic_Provides_for_Python_RPM_Packages). The spec file using name conversion based on automatic provides looks like [this](https://src.fedoraproject.org/rpms/pew/blob/master/f/pew.spec#_17).
